### DISCUSSION TITLE
Why most extension ideas fail

### DISCUSSION THREAD

**Discussion by ghost** [TurboWarp/extensions#647] [General]
Inspired by rdococ on Scratch.

There has been many issues, pull requests, and discussions about new unsandboxed extensions to be added on the gallery. Yet most of them have not been implemented, and many awesome suggestions are outright rejected. Here's why:

GarboMuffin has a **clear vision** of what they want Turbowarp to be. Turbowarp is designed to be cross compatible with Scratch; it is a Scratch mod that compiles projects with JavaScript. If every unsandboxed extension (which are not compatible with Scratch) were implemented, not only there would be a million extensions, new users will be confused to which why there's so many.

Flexibility and power are also goals, but they take a backseat to this "vision". Unsandboxed extensions are for experimenting and packaging. Any suggestion that you make has to be so good, that it outweighs the challenge of **new users facing it for the first time**, as well as outweighing these factors enough to justify **the weight of simply having an extra feature**.

A popular extension added to the gallery is Looks Plus, created by LilyMakesThings. (#405)

It expands on Scratch's Looks category, you can SVG skin sprites, get the number of costumes, layer, and more.
It used to be a pull request, and it took weeks to actually be added. The basic rules for submitting an extension is:

- It MUST not contain any bugs. GarboMuffin does not want to implement buggy extensions.
- You MUST not add blocks that are already in other extensions.
- It MUST not be a dangerous extension which could potentially harm one's device.

Both features' flexibility and usefulness for experimenting and packaging outweigh their complexity to a Turbowarp novice. This is not the case for the _vast majority_ of the features suggested in this discussion recently, and that's why they haven't been added to Turbowarp.

I'm very sorry if I cannot explain well. I was a bit tired while writing this. Have a good day.

**Comment by LilyMakesThings** [comment]
I think some of this is just wrong. First of all, I don't speak for Garbo, but this is what I think.

You'd be surprised how many extensions get accepted, actually. The majority of them do. From Garbo himself: 
_"We strive to be tolerant of accepting almost any extension, including one-use novelty extensions or extensions that are similar to ones that already exist."_

The reason why there are so many pull requests right now for extensions is really simple: Garbo has a life. He's the one that has to manually review every extension for bugs, and he has to make sure they work in the Interpreter as well as the Compiler (that's like 2 different TurboWarps!)

A lot more recent extensions (especially some of mine) tend to do things that are very experimental or haven't been tested thoroughly. They'll need more time before they're added.

**Reply by ghost** [reply]
Okay, that makes more sense now. Thanks for explaining Lily.
